src/components/Calculator.js

```diff
@@ -1,62 +1,49 @@
-import React, { Component } from 'react';
+import React, { useState } from 'react';
 // eslint-disable-next-line import/extensions
 import calculate from '../logic/calculate';
 import './Calculator.css';
 
-// eslint-disable-next-line react/prefer-stateless-function
-class Calculator extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      total: 0,
-      next: null,
-      operation: null,
-      err: null,
-    };
-    this.handleClick = this.handleClick.bind(this);
-  }
+const Calculator = () => {
+  const [state, setState] = useState({
+    total: 0, next: null, operation: null, err: null,
+  });
 
-  handleClick = (e) => {
-    this.setState((state) => calculate(state, e.target.textContent));
-  }
+  const handleClick = (e) => {
+    setState((prevState) => calculate(prevState, e.target.textContent));
+  };
 
-  render() {
-    const {
-      total, next, operation, err,
-    } = this.state;
-    return (
-      <div className="calculator">
-        <div className="calculator-display">
-          { next || operation || total || err || 0 }
+  return (
+    <div className="calculator">
+      <div className="calculator-display">
+        { state.next || state.operation || state.total || state.err || 0 }
+      </div>
+      <div className="container">
+        <div className="calculator-keypad">
+          <button type="button" onClick={handleClick} className="calculator-key key-clear">AC</button>
+          <button type="button" onClick={handleClick} className="calculator-key key-sign">+/-</button>
+          <button type="button" onClick={handleClick} className="calculator-key key-percent">%</button>
+          <button type="button" onClick={handleClick} className="calculator-key key-0">0</button>
+          <button type="button" onClick={handleClick} className="calculator-key key-dot">.</button>
+          <button type="button" onClick={handleClick} className="calculator-key key-1">1</button>
```

Comment:
Great job @OmarMWarraich 🤝

- [ ] [Optional] Kindly make sure your calculator monitor keeps track of the transactions that you enter into the calculator instead of only displaying the result.
![calc](https://user-images.githubusercontent.com/63917725/191680532-701f6c19-a269-4c27-8032-a42c4c1c0252.PNG)